app/models/competition.rb

```diff
@@ -2923,4 +2923,11 @@ def attempt_auto_close!
     threshold_reached = fully_paid_registrations_count >= auto_close_threshold && auto_close_threshold.positive?
     threshold_reached && update(closing_full_registration: true, registration_close: Time.now)
   end
+
+  def wcif_registrant_id_for(registration)
+    registrations.wcif_ordered.each.with_index(1) do |r, idx|
+      return idx if r.id == registration.id
+    end
+    nil
```

Comment:
I think Ruby has a stdlib method for that
```suggestion
    registrations.wcif_ordered.find_index { it.id == registration.id }
```

Reply:
I asked GPT for suggestions to handle the bulk registrations. and it came up with this